iree/turbine/kernel/_support/tracing.py

```diff
@@ -129,6 +132,8 @@ def create_arg(self, a):
             return a
         if isinstance(a, IndexMapping):
             return a
+        if isinstance(a, GenericDot):
```

Comment:
Could you describe the weird behavior that happens when you don't do this and add it as a TODO here? So this is not needed for GEMMs, but needed for Attention. TODO investigate further.

Reply:
This code is needed anyway as we are capturing something which is not a `fx.Node`, not sure why gemm worked without it, but I don't think we need any deep invesigation.